src/thing_list.c

```diff
@@ -690,7 +712,8 @@ long anywhere_thing_filter_call_bool_filter(const struct Thing *thing, MaxTngFil
 {
     if ((param->class_id == -1) || (thing->class_id == param->class_id))
     {
-        if ((param->model_id == -1) || (thing->model == param->model_id))
+        if((is_creature_match(thing, param->model_id) && param->class_id == TCls_Creature) 
+            || (param->model_id == -1) || (thing->model == param->model_id))
```

Comment:
I guess these 2 lines could go into a is_thing_match function, would be a short function but bit cleaner

and if any wildcards for other classes were made would have to be done anyways